src/index.ts

```diff
@@ -1,285 +1,144 @@
 import * as puppeteer from 'puppeteer';
 
+import { getMetamask } from './metamask';
 import downloader from './metamaskDownloader';
 
-export type LaunchOptions = Parameters<typeof puppeteer["launch"]>[0] & {
-  metamaskVersion?: string
-}
+// re-export
+export { getMetamask };
+
+export type LaunchOptions = Parameters<typeof puppeteer['launch']>[0] & {
+  metamaskVersion?: string;
+};
 
 export type MetamaskOptions = {
-  seed?: string
-  password?: string
-  extensionId?: string
-  extensionUrl?: string
-}
+  seed?: string;
+  password?: string;
+  extensionId?: string;
+  extensionUrl?: string;
+};
 
 export type Dappeteer = {
-  lock: () => Promise<void>
-  unlock: (password: string) => Promise<void>
-  addNetwork: (url: string) => Promise<void>
-  importPK: (pk: string) => Promise<void>
-  switchAccount: (accountNumber: number) => Promise<void>
-  switchNetwork: (network: string) => Promise<void>
-  confirmTransaction: (options?: TransactionOptions) => Promise<void>
-  sign: () => Promise<void>
-  approve: () => Promise<void>
-}
+  lock: () => Promise<void>;
+  unlock: (password: string) => Promise<void>;
+  addNetwork: (url: string) => Promise<void>;
+  importPK: (pk: string) => Promise<void>;
+  switchAccount: (accountNumber: number) => Promise<void>;
+  switchNetwork: (network: string) => Promise<void>;
+  confirmTransaction: (options?: TransactionOptions) => Promise<void>;
+  sign: () => Promise<void>;
+  approve: () => Promise<void>;
+};
 
 export type TransactionOptions = {
-  gas: number
-  gasLimit: number
-}
-
-export async function launch(pupeteerLib: typeof puppeteer, { args, ...rest }: LaunchOptions = {}): Promise<puppeteer.Browser> {
+  gas: number;
+  gasLimit: number;
+};
+
+export async function launch(
+  puppeteerLib: typeof puppeteer,
+  { args, ...rest }: LaunchOptions = {},
+): Promise<puppeteer.Browser> {
   // eslint-disable-next-line @typescript-eslint/naming-convention
   const METAMASK_PATH = await downloader(rest.metamaskVersion);
 
-  return pupeteerLib.launch({
+  return puppeteerLib.launch({
     headless: false,
-    args: [
-      `--disable-extensions-except=${METAMASK_PATH}`,
-      `--load-extension=${METAMASK_PATH}`,
-      ...(args || [])
-    ],
-    ...rest
-  })
+    args: [`--disable-extensions-except=${METAMASK_PATH}`, `--load-extension=${METAMASK_PATH}`, ...(args || [])],
+    ...rest,
+  });
 }
 
-export async function getMetamask(
+export async function setupMetamask(
```

Comment:
could you add comment on that method that it will setup metamask account. Maybe also change return type to be Metamask methods?